### PR TITLE
HTCONDOR-2369 make DCpermission heirarchy a class with only static me…

### PR DESCRIPTION
**Pull request by johnkn**
…mbers

   and make the hierarchy a static const table instead of code.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tools/ping.cpp [hunk 1]
```diff
@@ -68,6 +313,106 @@ void process_err_stack(CondorError *errstack) {
...
+						banner = nullptr;
+					}
+
+					printf("%s%s = %s\n", hash, param_name.c_str(), value.ptr());
+					if (param_name != from_param) fprintf(stdout, "%s # from: %s\n", hash, from_param.c_str());
+					reported.insert(from_param);
+
+					const MACRO_META * pmet = nullptr;
+					const char * def_val = nullptr;
+					const char * rawval = param_get_info(from_param.c_str(), subsys, nullptr, dotted_param, &def_val, &pmet);
```
Comment (GregThain) [inline src/condor_tools/ping.cpp:373]: Is this used anywhere?

File: src/condor_tools/ping.cpp [hunk 2]
```diff
@@ -68,6 +313,106 @@ void process_err_stack(CondorError *errstack) {
...
+			banner = nullptr;
+
+			formatstr(param_name, knob, PermString(DEFAULT_PERM));
+			if (subsys) { param_name += "_"; param_name += subsys; }
+
+			if (ends_with(knob, "_CRYPTO_METHODS")) {
+				std::string crypto_methods = SecMan::getDefaultCryptoMethods();
+				std::string filtered_methods = SecMan::filterCryptoMethods(crypto_methods);
+				if (filtered_methods != crypto_methods) {
+					fprintf(stdout, "%s = %s\n"
```
Comment (GregThain) [inline src/condor_tools/ping.cpp:396]: Are we missing an argument here?

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_tools/ping.cpp (hunk 1 above)
Comment (johnkn) [reply src/condor_tools/ping.cpp:373]: the MACRO_META data is used to get file and line number.

File: src/condor_tools/ping.cpp (hunk 2 above)
Comment (johnkn) [reply src/condor_tools/ping.cpp:396]: 3 %s and 3 args.  what am I missing?

Comment (johnkn) [reply src/condor_tools/ping.cpp:396]: I see it, an extraneous comma...

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**